### PR TITLE
Add initial support for modeling_xlm encoder on NPU

### PR DESCRIPTION
**Pull request by sgwhat**
### 1. Why the change?

Support running Bcembedding with ipex-llm optimization on NPU.

### 2. User API changes

We keep the same usage as Bcembedding:
```
from ipex_llm.transformers.npu_model import EmbeddingModel
model = EmbeddingModel(model_path,
                       torch_dtype=torch.float16,
                       optimize_model=True)
```

### 4. How to test?
- [x] Local Test

### CODE REVIEW

File: python/llm/example/NPU/HF-Transformers-AutoModels/Multimodal/README.md [hunk 1]
```diff
@@ -32,6 +33,9 @@ pip install torchvision
 # [optional] for MiniCPM-V-2_6
 pip install timm torch==2.1.2 torchvision==0.16.2
 
+# [optional] for Bce-Embedding-Base-V1
```
Comment (plusbang) [inline python/llm/example/NPU/HF-Transformers-AutoModels/Multimodal/README.md:36]: with ipex-llm[npu] installation option, transformers==4.40.0 is installed, no need here.

File: python/llm/example/NPU/HF-Transformers-AutoModels/Multimodal/README.md [hunk 2]
```diff
@@ -148,3 +152,23 @@ rtf_avg: 0.090: 100%|███████████████████
...
+```
+
+Arguments info:
+- `--repo-id-or-model-path REPO_ID_OR_MODEL_PATH`: argument defining the asr repo id for the model (i.e. `maidalun1020/bce-embedding-base_v1`) to be downloaded, or the path to the asr checkpoint folder.
+
+#### Sample Output
+##### [maidalun1020/bce-embedding-base_v1](https://huggingface.co/maidalun1020/bce-embedding-base_v1) |
+
+```log
+Inference time: 0.3975503444671631 s
```
Comment (plusbang) [inline python/llm/example/NPU/HF-Transformers-AutoModels/Multimodal/README.md:169]: Please use `Inference time: xxx s` instead.

File: python/llm/example/NPU/HF-Transformers-AutoModels/Multimodal/bce-embedding.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    )
+    parser.add_argument("--lowbit-path", type=str,
+        default="",
+        help="The path to the lowbit model folder, leave blank if you do not want to save. \
+            If path not exists, lowbit model will be saved there. \
+            Else, lowbit model will be loaded.",
+    )
+    parser.add_argument('--prompt', type=str, default="'sentence_0', 'sentence_1'",
+                        help='Prompt to infer')
+    parser.add_argument("--n-predict", type=int, default=32, help="Max tokens to predict")
```
Comment (plusbang) [inline python/llm/example/NPU/HF-Transformers-AutoModels/Multimodal/bce-embedding.py:46]: maybe we could remove this argument.

File: python/llm/src/ipex_llm/transformers/npu_models/xlm_mp.py [hunk 4]
```diff
@@ -0,0 +1,727 @@
...
+
+        return outputs
+
+
+def run_prefill(
+    model, max_output_len, max_prompt_len, transpose_value_cache, input_queue, result_queue
+):
+
+    layer_start = 0
+    layer_end = 12
```
Comment (plusbang) [inline python/llm/src/ipex_llm/transformers/npu_models/xlm_mp.py:422]: It maybe better to use model config to define.

File: python/llm/src/ipex_llm/transformers/npu_models/xlm_mp.py [hunk 5]
```diff
@@ -0,0 +1,727 @@
...
+            input_channels=input_channel,
+        )
+
+    def forward(self, hidden_states):
+        backend_cls = self.backend_cls_pooler
+        hidden_states = hidden_states.to(torch.float16)
+        return run_model(hidden_states, self.parameters, backend_cls, self.op_id)
+
+
+class LayerNorm(NNFactory):
```
Comment (plusbang) [inline python/llm/src/ipex_llm/transformers/npu_models/xlm_mp.py:653]: This is also used in minicpm-v, maybe we could move it to `common.py` as a common implementatin of `torch.nn.LayerNorm` later.

**Review by plusbang — APPROVED** [review]
other LGTM

File: python/llm/src/ipex_llm/transformers/npu_models/xlm_mp.py (hunk 5 above)
Comment (sgwhat) [reply python/llm/src/ipex_llm/transformers/npu_models/xlm_mp.py:653]: Good idea, will do this in next pr.

**Pull request merged**